test/unittest.jl

```diff
@@ -67,3 +67,7 @@ end
 @safetestset "Test hash of tree" begin
     include("test_hash.jl")
 end
+
+@safetestset "Test topology-preserving copy" begin
+    include("test_preserve_multiple_parents.jl")
+end
```

Comment:
Thanks for mentioning this! I just added a fix. Before this change, calling, e.g., `convert(Node{Float32}, tree)` on a tree would cause all shared children nodes to be duplicated (since `convert` has similar structure to `copy_node`, except for type conversions of the constants). Now this will work:

```julia
using Test
using SymbolicRegression

options = Options(;
    binary_operators=(+, -, *, ^, /, greater), unary_operators=(cos, exp, sin)
)
x1, x2, x3 = Node("x1"), Node("x2"), Node("x3")
base_tree = cos(x1 - 3.2 * x2) - x1^3.2
tree = sin(base_tree) + base_tree
# ^This tree has float64 constants

float32_tree = convert(Node{Float32}, tree)

# The linkage should be kept:
@test float32_tree.l.l === float32_tree.r
``` 
whereas before, this would fail.

Reply:
Thanks for elaborating. Looks good to me.

Reply:
Thanks!